packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -0,0 +1,403 @@
+  'Use the intrinsic `unknown` type for the catch callback variable.';
+
+export default createRule<[], MessageIds>({
+  name: 'use-unknown-in-catch-callback-variable',
+  meta: {
+    docs: {
+      description: 'Enforce typing arguments in .catch() callbacks as unknown',
+      requiresTypeChecking: true,
+    },
+    type: 'suggestion',
+    messages: {
+      useUnknown: useUnknownMessageBase,
+      useUnknownDestructuringPattern:
+        useUnknownMessageBase +
+        ' A destructuring pattern makes assumptions about the object that may not be valid.',
+      useUnknownSpreadArgs:
+        useUnknownMessageBase +
+        ' Your argument list may contain a handler that does not use `unknown` for the catch variable.',
+      addUnknownTypeAnnotationSuggestion:
+        'Add an explicit `: unknown` type annotation to the catch variable.',
+      addUnknownRestTypeAnnotationSuggestion:
+        'Add an explicit `: [unknown]` type annotation to the catch rest variable.',
+      wrongTypeAnnotationSuggestion:
+        'Change existing type annotation to `: unknown`',
+      wrongRestTypeAnnotationSuggestion:
+        'Change existing type annotation to `: [unknown]`',
+    },
+    fixable: 'code',
+    schema: [],
+    hasSuggestions: true,
+  },
+
+  defaultOptions: [],
+
+  create(context) {
+    const services = getParserServices(context);
+    const checker = services.program.getTypeChecker();
+
+    function isPromiseCatchAccess(node: TSESTree.Expression): boolean {
+      if (node.type !== AST_NODE_TYPES.MemberExpression) {
+        return false;
+      }
+
+      if (!isStaticMemberAccessOfValue(node, 'catch')) {
+        return false;
+      }
+
+      const object = node.object;
+
+      const objectTsNode = services.esTreeNodeToTSNodeMap.get(object);
+      const tsNode = services.esTreeNodeToTSNodeMap.get(node);
+      if (
+        !tsUtils.isThenableType(
+          checker,
+          tsNode,
+          checker.getTypeAtLocation(objectTsNode),
+        )
+      ) {
+        return false;
+      }
+
+      return true;
+    }
+
+    function isFlaggableHandlerType(type: ts.Type): boolean {
+      for (const unionPart of tsUtils.unionTypeParts(type)) {
+        const callSignatures = tsUtils.getCallSignaturesOfType(unionPart);
+        if (callSignatures.length === 0) {
+          // Ignore any nonfunction components to the type. Those are not this rule's problem.
+          continue;
+        }
+
+        for (const callSignature of callSignatures) {
+          const firstParam = callSignature.parameters.at(0);
+          if (!firstParam) {
+            // it's not an issue if there's no catch variable at all.
+            continue;
+          }
+
+          let firstParamType = checker.getTypeOfSymbol(firstParam);
+
+          // Deal with a rest arg.
+          // TODO PR QUESTION - is there a util for this? I just copied it from no-misused-promises.
+          const decl = firstParam.valueDeclaration;
```

Comment:
Question for maintainers - I more or less copied this pattern for dealing with rest args from no-misused-promsies:
https://github.com/typescript-eslint/typescript-eslint/blob/289ee8885ce26714296d8258abb12a2fa2c38d61/packages/eslint-plugin/src/rules/no-misused-promises.ts#L544-L589

Is there/should there be a shared utility to "go into rest arg"?